Sources/NIOSSL/SSLCallbacks.swift

```diff
@@ -52,6 +52,38 @@ public enum NIOSSLVerificationResult: Sendable {
     }
 }
 
+/// The result of an attempt to verify an X.509 certificate, with associated metadata if the certificate was successfully verified.
+public enum NIOSSLVerificationResultWithMetadata: Sendable, Hashable {
+    /// The certificate was successfully verified; the associated value contains metadata captured during verification.
+    case certificateVerified(VerificationMetadata)
+
+    /// The certificate was not verified.
+    case failed
+
+    /// The certificate was successfully verified with no metadata returned.
+    public static let certificateVerified: Self = .certificateVerified(.init())
```

Comment:
I'm not entirely convinced this is a helper we want, actually. Users seem unlikely to want to use the new API and then return nothing.